gdsfactory/cell.py

```diff
@@ -53,36 +54,42 @@ class Settings(BaseModel):
     child: dict[str, Any] | None = None
 
 
-def cell(func: _F) -> _F:
-    """Decorator for Component functions.
-
-    Implements a cache so that if a component has already been build
-    it will return the component from the cache directly.
+def cell(
+    func: _F | None = None,
+    /,
+    *,
+    autoname: bool = True,
+    max_name_length: int = CONF.max_name_length,
+    include_module: bool = False,
+    with_hash: bool = False,
+    ports_off_grid: str = CONF.ports_off_grid,
+    ports_not_manhattan: str = CONF.ports_not_manhattan,
+    flatten: bool = False,
+    naming_style: str = "default",
+    default_decorator: Callable[[Component], Component] | None = None,
+) -> Callable[[_F], _F]:
+    """Parametrized Decorator for Component functions.
+
+    decorator_kwargs are the default settings that the decorator will use, they can be overridden by the decorated function's kwargs.
+
+    Args:
+        autoname: True renames Component based on args and kwargs. True by default.
+        max_name_length: truncates name beyond some characters with a hash.
+        include_module: True adds module name to the cell name.
+        with_hash: True adds a hash to the cell name.
+        ports_off_grid: "warn", "error" or "ignore". Checks if ports are on grid.
+        ports_not_manhattan: "warn", "error" or "ignore". Checks if ports are manhattan.
+        flatten: False by default. True flattens component hierarchy.
+        naming_style: "default" or "updk". "default" is the default naming style.
+        default_decorator: default decorator to apply to the component. None by default.
+
+    kwargs:
+        cache: avoids creating duplicated Components.
+        name: names Components uniquely name based on parameters.
```

Comment:
There is no `**kwargs` argument above - these will be rejected if passed to `cell`.